src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -16,12 +13,19 @@ internal sealed class VisualStudioProjectFactory
     {
         private readonly VisualStudioWorkspaceImpl _visualStudioWorkspaceImpl;
         private readonly HostDiagnosticUpdateSource _hostDiagnosticUpdateSource;
+        private readonly ImmutableArray<Lazy<ISourceFileInfoProvider>> _sourceFileInfoProviders;
 
         [ImportingConstructor]
         // TODO: remove the AllowDefault = true on HostDiagnosticUpdateSource by making it a proper mock
-        public VisualStudioProjectFactory(VisualStudioWorkspaceImpl visualStudioWorkspaceImpl, [Import(AllowDefault = true)] HostDiagnosticUpdateSource hostDiagnosticUpdateSource)
+        // TODO: we might want to add metadata on ISourceFileInfoProvider to avoid unnecessary realization of service
```

Comment:
We will definitely require this - our assembly needs to be loaded in just web scenarios

Reply:
We already get mucho hate mail from them due to this editor thing that has been blocked for a year 😆  - I'd really like to stay off of their radar.

Reply:
:) sure.